src/sql/src/plan/statement/show.rs

```diff
@@ -722,6 +740,31 @@ impl<'a> ShowSelect<'a> {
     }
 }
 
+pub struct ShowColumnsSelect<'a> {
+    id: GlobalId,
+    new_resolved_ids: ResolvedIds,
+    show_select: ShowSelect<'a>,
+}
+
+impl<'a> ShowColumnsSelect<'a> {
+    pub fn describe(self) -> Result<StatementDesc, PlanError> {
+        self.show_select.describe()
+    }
+
+    pub fn plan(self) -> Result<Plan, PlanError> {
+        let select_plan = self.show_select.plan()?;
```

Comment:
I'm wondering if here you match against `select_plan` for `Plan::Select` and extract the inner `SelectPlan` struct, which would go inside of `ShowColumnsPlan` instead of a `Box<Plan>`. This way when sequencing the plan above, we don't need to loop or mutate the `plan`, we can sequence the select plan directly. What do you think?

Reply:
The reason for that is `dml::plan_select` can either return a Plan::Select or a Plan::SideEffectingFunc. Looking into it more, today side effecting functions are limited to topmost select (only `select pg_cancel_backend(1)` basically) which won't occur here because we construct the body of the select. 

I'll make your change and bail if we ever don't get a Plan::Select when planning.

Reply:
Nice! This looks great now, thanks for making the change!